omnigibson/objects/object_base.py

```diff
@@ -191,12 +190,13 @@ def _post_load(self):
             lazy.pxr.PhysxSchema.PhysxArticulationAPI.Apply(root_prim)
             self.self_collisions = self._load_config["self_collisions"]
 
-        # Update semantics
-        lazy.omni.isaac.core.utils.semantics.add_update_semantics(
-            prim=self._prim,
-            semantic_label=self.category,
-            type_label="class",
-        )
+        for link in self.links.values():
+            for vmesh in link.visual_meshes.values():
+                lazy.omni.isaac.core.utils.semantics.add_update_semantics(
+                    prim=vmesh._prim,
+                    semantic_label=self.category,
+                    type_label="class",
+                )
```

Comment:
FYI reviewer, this makes visual geoms that get detached from the object (e.g. in the case of particle system templates) also have the right ID, plus it prevents things like stains also ending up with the label of the object they are underneath.